sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/SourceOptions.scala

```diff
@@ -36,6 +36,9 @@ class SourceOptions(
   // A flag to always respect the Spark schema restored from the table properties
   val respectSparkSchema: Boolean = parameters
     .get(RESPECT_SPARK_SCHEMA).map(_.toBoolean).getOrElse(DEFAULT_RESPECT_SPARK_SCHEMA)
+
+  // Expected parallelism from the data source properties.
+  val parallel: Option[Int] = parameters.get(PARALLEL).map(_.toInt)
```

Comment:
Shall we verify the parameter? if PARALLEL is type of the `String` or any other that it is not `Int`, The will encounter a conversion error.